lib/qrolo.dart

```diff
@@ -27,526 +28,277 @@ import 'package:qrolo/src/html/media/utilities/is_media_device_camera_available.
-    // ignore: undefined_prefixed_name
-    ui.platformViewRegistry.registerViewFactory(
-      viewFactoryDivViewID,
-      (int id) => QRolo.videoDiv,
-    );
-
-    scanStream();
+    video = html.VideoElement();
+    // canvas = new html.CanvasElement(width: );
+    // ctx = canvas.context2D;
+    QRolo.vidDiv.children = [video!];
+    // ignore: UNDEFINED_PREFIXED_NAME
+    ui.platformViewRegistry
+        .registerViewFactory(viewID, (int id) => QRolo.vidDiv);
+    // initRenderers();
+    Timer(Duration(milliseconds: 500), () {
+      start();
+    });
   }
 
-  /// Methods should not be exposed
-  /// Maybe wrap this into the _web-only implementation as well
-  void startContinuousScanningLoop({
-    int scanIntervalMs = defaultScanIntervalMilliseconds,
-  }) {
-    /*
-      1. Start camera media stream
-        a. If null unable to capture then try again or error feedback dev/user
-      2. Play (add to video element)
-      3. Capture compare qr code (loop)
-    */
-    // - FIXME: Do not need to re-call once the reference is available
-    // Parts of the nested code flow should only be called once then...
-
-    Timer(
-      Duration(
-        milliseconds: scanIntervalMs,
-      ),
-      () {
-        scanStream();
-      },
-    );
+  void start() async {
+    await _makeCall();
+    // if (timer == null || !timer.isActive) {
+    //   timer = Timer.periodic(Duration(milliseconds: 500), (timer) {
+    //     if (code != null) {
+    //       timer.cancel();
+    //       Navigator.pop(context, code);
+    //       return;
+    //     }
+    //     _captureFrame2();
+    //     if (code != null) {
+    //       timer.cancel();
+    //       Navigator.pop(context, code);
+    //     }
+    //   });
+    // }
+    if (!widget.clickToCapture) {
+      // instead of periodic, which seems to have some timing issues, going to call timer AFTER the capture.
+      Timer(Duration(milliseconds: 200), () {
+        _captureFrame2();
+      });
+    }
   }
 
-  Future<String?> scanStream() async {
-    // 1. Start camera stream and get back
-    final videoStream = await callPlatformOpenMediaVideoStream();
-
-    if (videoStream == null) {
-      //
-      debugPrint('Error accessing camera stream getUserMedia()');
-
-      return null;
+  void cancel() {
+    if (timer != null) {
+      timer!.cancel();
+      timer = null;
     }
-    // Not assigned directly to allow local context null discrimination check
-    // Otherwise technically the _cameraStream could be nulled right before use
-    _cameraStream = videoStream;
-
-    // 1. a. Present the stream
-    final dynamic? playResult = await startPlayingStream(
-      videoStream,
-      videoElement,
-    );
-
-    var constraints = UserMediaOptions(
-        // audio: false,
-        video: VideoOptions(facingMode: "user"));
-    // dart style, not working properly:
-    // var stream =
-    //     await html.window.navigator.mediaDevices.getUserMedia(constraints);
```

Comment:
this still appears to return the given MediaStream for the camera video source

Not sure what side effects or mutations lead to it not working later.

May be a red herring old legacy comment